includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,297 @@
+        }
+
+        // Storing the plan id as product metadata, unique set to true
+        return $plan['id'];
+    }
+
+    protected function getPlanArguments($product)
+    {
+        $productId = $product['product_id'];
+
+        $period       = WC_Subscriptions_Product::get_period($productId);
+        $interval     = WC_Subscriptions_Product::get_interval($productId);
+        $recurringFee = WC_Subscriptions_Product::get_price($productId);
+
+        $salePrice    = WC_Subscriptions_product::get_meta_data($productId, 'sale_price', 0);
+
+        //
+        // If the item is on sale, sell it for the sale price
+        //
+        if (empty($salePrice) === false)
+        {
+            $recurringFee = $salePrice;
+        }
+
+        $planArgs = array(
+            'period'   => $this->getProductPeriod($period),
+            'interval' => $interval
+        );
+
+        $item = array(
+            'name'     => $product['name'],
+            'amount'   => (int) round($recurringFee * 100),
+            'currency' => get_woocommerce_currency(),
+        );
+
+        if ($item['currency'] !== self::INR)
+        {
+            $this->razorpay->handleCurrencyConversion($item);
+        }
+
+        $planArgs['item'] = $item;
+
+        return $planArgs;
+    }
+
+    protected function getProductPeriod($period)
+    {
+        $periodMap = array(
+            'day'   => 'daily',
+            'week'  => 'weekly',
+            'month' => 'monthly',
+            'year'  => 'yearly'
+        );
+
+        return $periodMap[$period];
+    }
+
+    protected function getCustomerId($order)
+    {
+        $data = $this->getCustomerInfo($order);
+        $data['fail_existing'] = '0';
+
+        try
+        {
+            $customer = $this->api->customer->create($data);
+        }
+        catch (Exception $e)
+        {
+            $message = $e->getMessage();
+            return 'RAZORPAY ERROR: Customer creation failed with the message \'' . $message . '\'';
+        }
+
+        return $customer['id'];
+    }
+
+    protected function getCustomerInfo($order)
+    {
+        if (version_compare(WOOCOMMERCE_VERSION, '2.7.0', '>='))
+        {
+            $args = array(
+                'name'    => $order->get_billing_first_name() . ' ' . $order->get_billing_last_name(),
+                'email'   => $order->get_billing_email(),
+                'contact' => $order->get_billing_phone(),
+            );
+        }
+        else
+        {
+            $args = array(
+                'name'    => $order->billing_first_name . ' ' . $order->billing_last_name,
+                'email'   => $order->billing_email,
+                'contact' => $order->billing_phone,
+            );
+        }
+
+        return $args;
+    }
+
+    public function getProduct($order)
+    {
+        $products = $order->get_items();
+
+        // Technically, subscriptions work only if there's one array in the cart
+        if (sizeof($products) > 1)
```

Comment:
@ccrims0n I don't get how it's easier to understand. Small issue, can we leave as is?